app/src/main/java/com/nervousfish/nervousfish/activities/CustomKeyboardHelper.java

```diff
@@ -0,0 +1,199 @@
+import android.inputmethodservice.KeyboardView;
+import android.support.v7.widget.AppCompatEditText;
+import android.text.Editable;
+import android.view.ActionMode;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.Window;
+import android.view.WindowManager;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+
+import com.nervousfish.nervousfish.R;
+
+import java.util.Objects;
+
+import static android.content.Context.INPUT_METHOD_SERVICE;
+
+/**
+ * A helper method for {@link Activity}s that want to use a custom keyboard.
+ */
+final class CustomKeyboardHelper {
+
+    private final Activity activity;
+    private final KeyboardView keyboardView;
+
+    /**
+     * Creates a new {@link CustomKeyboardHelper} which inserts a custom keyboard into an activity.
+     *
+     * @param activity The {@link Activity} where the custom keyboard should be used.
+     */
+    CustomKeyboardHelper(final Activity activity) {
+        this.activity = activity;
+        this.keyboardView = (KeyboardView) activity.findViewById(R.id.keyboard_view);
+
+        // Create custom keyboard layout
+        final Keyboard keyboard = new Keyboard(activity, R.xml.qwerty);
+        this.keyboardView.setKeyboard(keyboard);
+        this.keyboardView.setPreviewEnabled(false);
+        this.keyboardView.setOnKeyboardActionListener(new CustomKeyboardHelper.OnCustomKeyboardActionListener(activity));
+
+        // Disable the default keyboard
+        final Window window = this.activity.getWindow();
+        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
+    }
+
+    /**
+     * Add a new input that should use the custom keyboard.
+     *
+     * @param editText The {@link EditText} that should use the keyboard.
+     */
+    void addInput(final EditText editText) {
+        editText.setOnFocusChangeListener(this::focusListener);
+        editText.setOnClickListener(this::clickListener);
+        editText.setCustomSelectionActionModeCallback(new CustomKeyboardHelper.EditPasswordSelectionCallback());
+    }
+
+    /**
+     * Find out if the custom keyboard is visible.
+     *
+     * @return A {@code boolean} indicating whether the custom keyboard is visible
+     */
+    boolean isVisible() {
+        return this.isCustomKeyboardVisible();
+    }
+
+    /**
+     * Hide the custom keyboard.
+     */
+    void hide() {
+        this.hideCustomKeyboard();
+    }
+
+    private void focusListener(final View view, final Boolean hasFocus) {
+        if (hasFocus) {
+            this.clickListener(view);
+        } else {
+            this.hideCustomKeyboard();
+        }
+    }
+
+    private void clickListener(final View view) {
+        this.keyboardView.setVisibility(View.VISIBLE);
+        this.keyboardView.setEnabled(true);
+        this.hideRealKeyboard(view);
+    }
+
+    private void hideRealKeyboard(final View view) {
```

Comment:
Oops 😅 put it there while trying to get the custom keyboard working in the `CreateProfileActivity` but forgot to remove it